backends-velox/src/main/scala/org/apache/spark/sql/expression/UDFResolver.scala

```diff
@@ -80,17 +96,129 @@ object UDFResolver extends Logging {
     logInfo(s"Registered UDF: $name -> $t")
   }
 
+  def parseName(name: String): (String, String) = {
+    val index = name.lastIndexOf("#")
+    if (index == -1) {
+      (name, Paths.get(name).getFileName.toString)
+    } else {
+      (name.substring(0, index), name.substring(index + 1))
+    }
+  }
+
+  def getFilesWithExtension(directory: java.nio.file.Path, extension: String): Seq[String] = {
+    Files
+      .walk(directory)
+      .iterator()
+      .asScala
+      .filter(p => Files.isRegularFile(p) && p.toString.endsWith(extension))
+      .map(p => p.toString)
+      .toSeq
+  }
+
+  def resolveUdfConf(conf: java.util.Map[String, String]): Unit = {
+    if (isDriver) {
+      if (localLibraryPaths != null) {
+        conf.put(BackendSettings.GLUTEN_VELOX_UDF_LIB_PATHS, localLibraryPaths)
+      }
+    } else {
+      val sparkConf = SparkEnv.get.conf
+      Option(conf.get(BackendSettings.GLUTEN_VELOX_UDF_LIB_PATHS)) match {
+        case Some(libs) =>
+          conf.put(
+            BackendSettings.GLUTEN_VELOX_UDF_LIB_PATHS,
+            getAllLibraries(libs, sparkConf, canAccessSparkFiles = true))
+        case None =>
+      }
+    }
+  }
+
+  // Try to unpack archive. Throws exception if failed.
+  def unpack(source: File, destDir: File): File = {
+    val sourceName = source.getName
+    val dest = new File(destDir, sourceName)
+    logInfo(
+      s"Unpacking an archive $sourceName from ${source.getAbsolutePath} to ${dest.getAbsolutePath}")
+    try {
+      Utils.deleteRecursively(dest)
+      Utils.unpack(source, dest)
+    } catch {
+      case e: Exception =>
+        throw new GlutenException(s"Unpack ${source.toString} failed.", e)
```

Comment:
nit: `Unpack ${source.toString} failed, please check if it is a compressed file`